### ISSUE TITLE
Add an option to quit the game when in a map

### ISSUE THREAD

**Issue by VortexParadox** [momentum-mod/game#2009]
### What feature is your improvement idea related to?

I believe it would be a great user experience improvement to add a method to be able to select Quit the game instead of having to click the same Quit button twice.

### Describe the solution you'd like

A method to allow quiting the game while in a map

### Describe alternatives you've considered, if any.

It could be done by
1. Adding a dropdown buttton to select quit game
2. Adding a system where you have to activate the quit button to exit the game E.g.:
    2.1. Holding the button to quit the game
    2.2. Holding and sliding the button to exit the game
3. Adding a popup to choose quit map or quit game to desktop (seems most sensible and removes the need to have a safeguard I'd say)

**Comment by Gocnak** [comment]
I think there was discussion to make it right clicking the quit button always does quit game, left click is more context aware

**Comment by VortexParadox** [comment]
I'm not sure right clicking a button is something users do on the regular to know it does anything. It's like the radio buttons on the map list, no one does that and you end up having to interrupt the player to teach something very specific

**Comment by Gocnak** [comment]
Yes, but so would click and hold, clicking and sliding, etc. Pop-up every time would get users annoyed, the safeguard is specifically only when in a run under the conditions the player sets.

Drop-down might work but I need to see another game doing that / some mockups.

**Comment by tsa96** [comment]
We've been discussing this whilst discussing safeguards recently. The plan was to allow you exit a map without warning if you're not in a run (and you can disable that safeguard), then to have the "exit game" from main menu be a "don't show me again popup" (though doing that popup is hard because needs some C++ refactor of UiToolkitAPI).

So in practice it would be two single clicks to edit, and the server shuts down and boots you to main menu almost instantly. Hopefully this is fast enough to people in practice, though honestly I'm not sure it'll satisfy everyone. I'm unsure of better alternatives in the UI though, we could do the right-click behaviour very easily but I agree with Vortex, it's hard to teach and pretty unintuitive.

**Comment by tsa96** [comment]
Thinking about it, yeah just not fussing with safeguards and having the quit map button trigger a three option popup maybe makes the most sense. In theory it is slightly more effort to close the game if you have the quit game popup disabled, since you have to most your mouse over to the center of the screen to press the buttons, but it's much more like other games and probably more intuitive.

**Comment by VortexParadox** [comment]
Yeah for most of the games I've seen that only have one "Leave" button immediately, it's clicking "Leave/Quit"  then a popup or screen that prompts them to choose between "Quit to Menu" and "Quit to Desktop". The popup could even be a slide down from the upper right of the screen, in order to minimize mouse movement.

**Comment by Gocnak** [comment]
Hmm yeah honestly with quitting being a big deal run wise it could be good to ensure people want to do it regardless of timer state. I think pop-up can work best here then.

**Comment by tsa96** [comment]
Okay great, this is an easy change. Just JS code in mainmenu.js to call a ShowPopupThreeOptions or whatever it's called if in pause menu state.
